client/src/components/MistakeOptionsMenu.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+  <b-dropdown
+    dropleft
+    variant="link"
+    no-caret>
+    <template #button-content>
+      <fa-icon
+        :icon="['fas', 'ellipsis-v']" />
+    </template>
+    <b-dropdown-item :to="'/journal/mistakes/edit/' + mistakeId">
+      <span class="mj-options">
+        <fa-icon
+          :icon="['fas', 'pen']" />
+        <span class="mj-options-text">
+          {{ $t('MistakeOptions.Edit') }}
+        </span>
+      </span>
+    </b-dropdown-item>
+    <b-dropdown-item @click="deleteMistake">
+      <span class="mj-options">
+        <fa-icon
+          :icon="['far', 'trash-alt']" />
+        <span class="mj-options-text">
+          {{ $t('MistakeOptions.Delete') }}
+        </span>
+      </span>
+    </b-dropdown-item>
+  </b-dropdown>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { MistakesActions } from '@/store/mistakes-module/actions';
+
+export default Vue.extend({
+  name: 'MistakeOptionsMenu',
+  props: {
+    mistakeId: {
+      required: true,
+      type: String,
+    },
+  },
+  methods: {
+    async deleteMistake(): Promise<void> {
+      await this.$store.dispatch(MistakesActions.Delete, this.mistakeId);
```

Comment:
When you are on mistakes list and click "Delete" while gaving mistake's solution page open, it stays opened but mistake is no longer existing - I would redirect to mistakes list after this action (to close the view).